### ISSUE TITLE
0.20.1 incompatible with no-sats 0.20.0 index made with --index-addresses

### ISSUE THREAD

**Issue by gmart7t2** [ordinals/ord#3981]
I made my ord index using ord 0.20.0 with --index-addresses but no --index-sats

Then I upgraded to ord 0.20.1

When I try to run the ord server I get: 

```
[2024-10-04T17:43:52Z INFO  ord::index::updater] Block 864134 at 2024-10-04 14:51:28 UTC with 2767 transactions…
thread '<unnamed>' panicked at src/index/updater.rs:564:15:
assertion failed: !self.index.have_full_utxo_index()
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

The --index-addresses will have caused 0.20.1 to index from block 0, but doesn't do the same for 0.20.0, so my 0.20.0 index doesn't have a full UTXO index, but 0.20.1 thinks it should.

Maybe 0.20.1 should be called 0.21.0 since it breaks index compatibility like this.

**Comment by so7ow** [comment]
Seeing same here

**Comment by gmart7t2** [comment]
Allowing the user to specify the first block to be indexed would fix this. 

Sometimes I want to build a quick index by skipping all but the most recent 10 blocks. That used to be possible with the flag that was removed in the 0.20.1 release. I don't always want the address index to be from block 0. I know my wallet was created 10 blocks ago, so it doesn't have any utxos from before then

**Comment by ep150de** [comment]
I upgraded my client from 0.19 to 0.20.1 and got a similar error:

index.redb` appears to have been built with an older, incompatible version of ord, consider deleting and rebuilding the index: index schema 25, ord schema 28

**Comment by so7ow** [comment]
> I upgraded my client from 0.19 to 0.20.1 and got a similar error:
> 
> index.redb` appears to have been built with an older, incompatible version of ord, consider deleting and rebuilding the index: index schema 25, ord schema 28

Expected. Upgrading major versions (like from 19.x to 20.x) generally requires a reindex.

**Comment by gmart7t2** [comment]
Looks like this is fixed in v21.0

https://x.com/rodarmor/status/1845160074123379018

**Comment by raphjaph** [comment]
Yes, thanks for opening this issue and sorry for the oversight on our end!
